### PR TITLE
feat: user role pagination

### PR DESCRIPTION
**Pull request by kurokn**
- used new interface `RequestAllResult` to wrap two outputs into one result array
- hidden the page limit selection box
- add `page` and `limit` as new parameters

### CODE REVIEW

File: app/components/Table.tsx [hunk 1]
```diff
@@ -107,6 +107,7 @@ interface Props {
   onPrev?: (val: number) => void;
   onNext?: (val: number) => void;
   loading?: boolean;
+  pageLimitBox?: string;
```
Comment (thegentlemanphysicist) [inline app/components/Table.tsx:110]: this won't be needed if we have that conditional

File: lambda/app/src/keycloak/users.ts [hunk 2]
```diff
@@ -28,16 +32,24 @@ export const searchUsers = async ({
...
-    username,
-    firstName,
-    lastName,
-    email,
-    attributes,
-  }));
+  const countRow = await kcAdminClient.users.find({ realm: 'standard', ...query, max: undefined, first: undefined });
+  const result: { count: number; rows: any[] } = {
+    count: countRow.length,
+    rows: users.map(({ username, firstName, lastName, email, attributes }) => ({
```
Comment (thegentlemanphysicist) [inline lambda/app/src/keycloak/users.ts:43]: Sorting would have to take place here, if you sort on the frontend you will have problems on multible page results

**Review by thegentlemanphysicist — COMMENTED** [review]
please confirm that there are no console log errors when you run locally

File: app/page-partials/my-dashboard/UserRoles.tsx [hunk 3]
```diff
@@ -415,14 +429,27 @@ const UserRoles = ({ selectedRequest, alert }: Props) => {
...
+                setSearchKey(val);
+                setPage(1);
+              }}
+              onEnter={(val) => {
+                setSearchKey(val);
+                setPage(1);
+              }}
               loading={loading}
+              rowCount={count}
+              onPrev={setPage}
```
[the remark this answers is not in the document]
Comment (kurokn) [reply app/page-partials/my-dashboard/UserRoles.tsx:447]: previous and next both works, compared with default one-list data, no data lost with the previous and next feature

File: app/components/Table.tsx (hunk 1 above)
Comment (kurokn) [reply app/components/Table.tsx:110]: removed

File: app/page-partials/my-dashboard/UserRoles.tsx [hunk 4]
```diff
@@ -415,14 +429,27 @@ const UserRoles = ({ selectedRequest, alert }: Props) => {
...
+              onNext={setPage}
               totalColSpan={20}
               searchColSpan={8}
               filterColSpan={12}
+              pageLimitBox={'hidden'}
```
[the remark this answers is not in the document]
Comment (kurokn) [reply app/page-partials/my-dashboard/UserRoles.tsx:452]: removed

**Pull request closed without merging**